scilpy/reconst/afd_along_streamlines.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from dipy.data import get_sphere
+import numpy as np
+from scipy.special import lpn
+
+from scilpy.reconst.utils import find_order_from_nb_coeff, get_b_matrix
+from scilpy.tractanalysis.grid_intersections import grid_intersections
+
+
+def afd_map_along_streamlines(sft, fodf, fodf_basis, length_weighting):
+    """
+    Compute the mean Apparent Fiber Density (AFD) and mean Radial fODF
+    (radfODF) maps along a bundle.
+
+    Parameters
+    ----------
+    sft : StatefulTractogram
+        StatefulTractogram containing the streamlines needed.
+    fodf : np.ndarray
+        fODF with shape (X, Y, Z, #coeffs)
+        coeffs depending on the sh_order
+    fodf_basis : string
+        Has to be descoteaux07 or tournier07
+    length_weighting : bool
+        If set, will weigh the AFD values according to segment lengths
+
+    Returns
+    -------
+    afd_sum : np.array
+        AFD map (weighted if length_weighting)
+    rd_sum : np.array
+        rdAFD map (weighted if length_weighting)
+    """
+
+
+    afd_sum, rd_sum, weights = \
+        afd_and_rd_sums_along_streamlines(sft, fodf, fodf_basis,
+                                          length_weighting)
+
+    non_zeros = np.nonzero(afd_sum)
+    weights_nz = weights[non_zeros]
+    afd_sum[non_zeros] /= weights_nz
+    rd_sum[non_zeros] /= weights_nz
+
+    return afd_sum, rd_sum
+
+
+def afd_and_rd_sums_along_streamlines(sft, fodf, fodf_basis,
+                                      length_weighting):
+    """
+    Compute the mean Apparent Fiber Density (AFD) and mean Radial fODF (radfODF)
+    maps along a bundle.
+
+    Parameters
+    ----------
+    sft : StatefulTractogram
+        StatefulTractogram containing the streamlines needed.
+    fodf : np.ndarray
+        fODF with shape (X, Y, Z, #coeffs).
+        #coeffs depend on the sh_order.
+    fodf_basis : string
+        Has to be descoteaux07 or tournier07.
+    length_weighting : bool
+        If set, will weigh the AFD values according to segment lengths.
+
+    Returns
+    -------
+    afd_sum_map : np.array
+        AFD map.
+    rd_sum_map : np.array
+        fdAFD map.
+    weight_map : np.array
+        Segment lengths.
+    """
+
+    sft.to_vox()
+    sft.to_corner()
+
+    fodf_data = np.asanyarray(fodf.dataobj)
```

Comment:
Why not` fodf.get_fdata() `?

Reply:
We had a discussion about this with @GuillaumeTh I think it's safer to get the data that way.